### DISCUSSION TITLE
How verify token and store secret ?

### DISCUSSION THREAD

**Discussion by innovaweb-dev** [panva/jose#211] [Q&A] [answered]
How verify a token ?
How and where store the secret ?

```` js
// login
	const {generateSecret} = require('jose/util/generate_secret')
			const secret = await generateSecret('HS256')
			console.log(secret);
			
			const accessToken = await new SignJWT({'userID': userID})
				.setProtectedHeader({alg: 'HS256'})
				.setIssuedAt()
				.setSubject('auth-access')
				.setIssuer('foo')
				.setExpirationTime(process.env.JWT_ACCESS_EXPIRATION * 100)
				.sign(secret)
								
			res.cookie('accessToken', accessToken, {
				httpOnly: true,
				secure: false, //TODO process.env.NODE_ENV === 'production'
				maxAge: process.env.JWT_ACCESS_EXPIRATION * 1000
			});


````
I try to verify access token, but does not work 
````js

// auth middleware
const secret = await generateSecret('HS256');
const { payload, protectedHeader }  = await  jwtVerify(accessToken, secret).promise((err)=> {
     console.log(err); // JWSSignatureVerificationFailed: signature verification failed
})
````

**Comment by panva** [comment, marked as answer]
> I try to verify access token, but does not work

Well, why are you verifying with a new random secret? Of course the signature validation is going to fail. Use the same secret for signing and verifying. Store the secret through whatever means fits your deployment. The secret can be passed in as a [Buffer or a KeyObject](https://github.com/panva/jose/blob/main/docs/types/types.keylike.md#type-alias-keylike) instance, both of which have APIs to (de)serialize the random value and come as standard APIs in Node.js
